tests/integration/security/https_jwt/https_jwt_test.go

```diff
@@ -0,0 +1,150 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"istio.io/istio/pkg/test/echo/common/response"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echotest"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/pkg/test/util/yml"
+	"istio.io/istio/tests/common/jwt"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/authn"
+)
+
+const (
+	authHeaderKey = "Authorization"
+)
+
+// TestJWTHTTPS tests the requestauth policy with https jwks server.
+func TestJWTHTTPS(t *testing.T) {
+	payload1 := strings.Split(jwt.TokenIssuer1, ".")[1]
+
+	framework.NewTest(t).
+		Features("security.authentication.jwt").
+		Run(func(t framework.TestContext) {
+			ns := apps.Namespace1
+			istioSystemNS := istio.ClaimSystemNamespaceOrFail(t, t)
+			args := map[string]string{"Namespace": istioSystemNS.Name()}
+			applyYAML := func(filename string, ns namespace.Instance) []string {
+				policy := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, filename))
+				for _, cluster := range t.AllClusters() {
+					t.ConfigKube(cluster).ApplyYAMLOrFail(t, ns.Name(), policy...)
+				}
+				return policy
+			}
+
+			jwtServer := applyYAML("../../../../samples/jwt-server/jwt-server.yaml", istioSystemNS)
```

Comment:
nit: `filepath.Join(env.IstioSrc,...`, so its not dependant on file structure

Reply:
I'm sorry but I couldn't understand the comment. Could you please elaborate it little bit.